packages/react-components/react-nav-preview/src/components/NavGroup/useNavGroup.styles.ts

```diff
@@ -0,0 +1,245 @@
+    lineHeight: tokens.lineHeightBase300,
+    outlineStyle: 'none',
+    position: 'relative',
+    ...shorthands.overflow('hidden'),
+    textTransform: 'none',
+  },
+});
+/* eslint-enable @typescript-eslint/naming-convention */
+
+/**
+ * Focus styles for the root slot
+ */
+const useFocusStyles = makeStyles({
+  // navGroup creates a custom focus indicator because the default focus indicator
+  // is applied using an ::after pseudo-element on the root. Since the selection
+  // indicator uses an ::after pseudo-element on the root, there is a conflict.
+  base: createCustomFocusIndicatorStyle(
+    {
+      ...shorthands.borderColor('transparent'),
+      outlineWidth: tokens.strokeWidthThick,
+      outlineColor: 'transparent',
+      outlineStyle: 'solid',
+      boxShadow: `
+      ${tokens.shadow4},
+      0 0 0 ${tokens.strokeWidthThick} ${tokens.colorStrokeFocus2}
+    `,
+      zIndex: 1,
+    },
+    { enableOutline: true },
+  ),
+});
+
+/** Indicator styles for when pending selection */
+const usePendingIndicatorStyles = makeStyles({
+  base: {
+    ':hover::before': {
+      backgroundColor: tokens.colorNeutralStroke1Hover,
+      ...shorthands.borderRadius(tokens.borderRadiusCircular),
+      content: '""',
+      position: 'absolute',
+    },
+    ':active::before': {
+      backgroundColor: tokens.colorNeutralStroke1Pressed,
+      ...shorthands.borderRadius(tokens.borderRadiusCircular),
+      content: '""',
+      position: 'absolute',
+    },
+    '@media (forced-colors: active)': {
+      ':hover::before': {
+        backgroundColor: 'Highlight',
+      },
+      ':active::before': {
+        backgroundColor: 'Highlight',
+      },
+    },
+  },
+});
+
+const useActiveIndicatorStyles = makeStyles({
+  base: {
+    '::after': {
+      backgroundColor: tokens.colorTransparentStroke,
+      ...shorthands.borderRadius(tokens.borderRadiusCircular),
+      content: '""',
+      position: 'absolute',
+      zIndex: 1,
+    },
+  },
+  selected: {
+    '::after': {
+      backgroundColor: tokens.colorCompoundBrandStroke,
+    },
+    ':hover::after': {
+      backgroundColor: tokens.colorCompoundBrandStrokeHover,
+    },
+    ':active::after': {
+      backgroundColor: tokens.colorCompoundBrandStrokePressed,
+    },
+    '@media (forced-colors: active)': {
+      '::after': {
+        backgroundColor: 'ButtonText',
+      },
+      ':hover::after': {
+        backgroundColor: 'ButtonText',
+      },
+      ':active::after': {
+        backgroundColor: 'ButtonText',
+      },
+    },
+  },
+});
+
+/**
+ * Styles for the icon slot.
+ */
+const useIconStyles = makeStyles({
+  base: {
+    gridColumnStart: 1,
+    gridRowStart: 1,
+    alignItems: 'center',
+    display: 'inline-flex',
+    justifyContent: 'center',
+    ...shorthands.overflow('hidden'),
+    [`& .${iconClassNames.filled}`]: {
+      display: 'none',
+    },
+    [`& .${iconClassNames.regular}`]: {
+      display: 'inline',
+    },
+  },
```

Comment:
These could move into a makeResetStyles call. The benefit is that there are fewer atomic css classes added to the slot, which improves perf and bundle size.

Reply:
✅